NexusMods.Archives.Nx/FileProviders/FileData/MemoryMappedFileData.cs

```diff
@@ -89,4 +110,29 @@ private unsafe void InitEmpty()
 
     /// <inheritdoc />
     ~MemoryMappedFileData() => Dispose();
+
+    #region Memory Access Hints for OSes
+
+#if NET5_0_OR_GREATER
+    // POSIX Compatible
+    [DllImport("libc.so.6", EntryPoint = "madvise")]
+    private static extern unsafe int madvise(byte* addr, nuint length, int advice);
```

Comment:
Probably needs changing to `libSystem.dylib` on macOS and unused for BSD as we can't verify that code works correctly.